modules/stochastic_tools/src/samplers/AdaptiveImportanceSampler.C

```diff
@@ -143,6 +151,10 @@ AdaptiveImportanceSampler::computeSample(dof_id_type /*row_index*/, dof_id_type
       }
       _prev_value[i] = (Normal::quantile(getRand(_step), _mean_sto[i], _std_factor * _std_sto[i]));
     }
+
+    // check if we have performed all the importance sampling steps
+    if (_step >= _num_samples_train + _num_importance_sampling_steps)
+      _is_sampling_completed = true;
```

Comment:
Maybe a question for @somu15 . Do we want to skip all this stuff in `computeSample` if `_is_sampling_completed == true`? Same question for PSS.

Reply:
I'm interested in the answer to. I did add a bunch of skips at some point in `Sampler.C` based on `_is_sampling_completed` but I didn't see much of a difference so I didn't leave it in the final PR. But if @somu15 thinks it would be better (especially for large simulations), I am fine putting it back.

Reply:
Does having `_is_sampling_completed = _step >= _num_samples_train + _num_importance_sampling_steps;` towards the end of `ComputeSample` launch one more step of simulations even when the total required simulations are met?

Reply:
So, I didn't get why for large simulations this may not be preferable.

Reply:
Just asking in case there's something I'm missing ;) but it sounds like there might not be anything to worry about

Reply:
I think so too

Reply:
Should it throw an error if we request a sample after the sampling is complete? Or return whatever the last sample was? Or move onto the next sample? Or reset the chain?

My inkling is to throw an error. But something should happen and it needs be well defined.

Reply:
Errors added to both samplers